Formula/libxml2.rb

```diff
@@ -41,7 +41,7 @@ class Libxml2 < Formula
 
   keg_only :provided_by_osx
 
-  depends_on :python => :optional
+  depends_on :python if MacOS.version <= :snow_leopard
```

Comment:
I don't really get what the problem is in relation to bindings, to be honest. My existing `libxml2` bindings can be imported by both system & Homebrew Python without any issues, and I see no way this PR changes that.